django_short_urls/views.py

```diff
@@ -58,15 +57,17 @@ def main(request, path):
 
     link = Link.find_by_hash(path)
 
+    redirect_suffix = None
+
     if link is None:
-        # Try to find a matching short link by removing valid "catchall" suffixes
-        path_prefix, redirect_suffix = suffix_catchall.get_hash_from(path)
+        # Try to find a matching prefix
+        parts = path.rsplit('/', 1)
```

Comment:
`rsplit` or `lsplit`? Do we ever create URLs with more than two dash-separated components?

Reply:
We can, yes, sometimes – at least for manually created urls – also, users could send broken stuff.

Reply:
In that case, why would we try to find a prefix in the right-most slashed part?

Reply:
Why would we want to try the left most one? We should remove suffixes one by one.

Reply:
It's not recursive though, we make a single attempt. To clarify my thoughts, this doesn't make sense to me:
```
path = "CompanySlug/AJobRef/garbage/junk"
parts = path.rsplit('/', 1)
# parts[0] == "CompanySlug/AJobRef/garbage"
Link.find_by_hash(parts[0])
```
Maybe that's an edge case (paths with >2 parts) we don't care about, but the use of `rsplit` is oddly specific, and I don't have a spec or test to validate it against :-)